### ISSUE TITLE
Exception from benchmark runner killed the GUI

### ISSUE THREAD

**Issue by F1Rumors** [nicehash/NiceHashMiner#1538]
GUI version: NiceHashMiner v1.9.0.22

The GUI exited when starting up - there was an unbenchmarked item, I permitted it to benchmark before starting, and it apparently died as as consequence. The logfile reported an exception:

```exception
[2019-05-23 08:47:26] [INFO] [BenchmarkHandler-GPU#3 MSI GeForce GTX 1070 Ti] Exception System.InvalidOperationException: Invoke or BeginInvoke cannot be called on a control until the window handle has been created.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at NiceHashMiner.Forms.Form_Benchmark.AddToStatusCheck(ComputeDevice device, Algorithm algorithm)
   at NiceHashMiner.Benchmarking.BenchmarkHandler.<Benchmark>d__12.MoveNext()
```

**Comment by S74nk0** [comment]
@F1Rumors please use [1.9.1.4](https://github.com/nicehash/NiceHashMinerLegacy/releases/tag/1.9.1.4). It is being actively developed and should have less issues and it will replace the old 1.9.0.xx versions completely.

**Comment by F1Rumors** [comment]
Understood - this is the purpose of the alpha, I agree. Installing 1.9.1.4 - will test that, too. thank you.

**Comment by F1Rumors** [comment]
Latest version also has issues -- everything went wrong after it started mining Beam on GMiner and apparently things did not go to plan. From the logs:

```csharp
[2019-06-08 12:47:40] [INFO] [GMiner-MINER_ID(0)-DEVICE_IDs(2)] Setting End and Stopping
[2019-06-08 12:47:40] [INFO] [Miner33-GMiner-algo(GrinCuckaroo29)-devs(NVIDIA:2)] Stop miner called
[2019-06-08 12:47:40] [INFO] [Miner33-GMiner-algo(GrinCuckaroo29)-devs(NVIDIA:2)] Stop miner called: STOP TASKS IN LIST: 1
[2019-06-08 12:47:40] [INFO] [Miner34-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0)] Stop miner called
[2019-06-08 12:47:40] [INFO] [Miner34-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0)] Stop miner called: STOP TASKS IN LIST: 1
[2019-06-08 12:47:46] [INFO] [NBMiner-MINER_ID(0)-DEVICE_IDs(0)] Setting End and Stopping
[2019-06-08 12:47:46] [INFO] [Miner34-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0)] Stop miner called
[2019-06-08 12:47:46] [INFO] [Miner34-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0)] Stop miner called: STOP TASKS IN LIST: 1
[2019-06-08 12:47:46] [INFO] [GMiner-MINER_ID(0)-DEVICE_IDs(2)] NEW MINER CREATED
[2019-06-08 12:47:46] [INFO] [Miner35-GMiner-algo(Beam)-devs(NVIDIA:2)] Starting miner watchdog task
[2019-06-08 12:47:46] [INFO] [NBMiner-MINER_ID(0)-DEVICE_IDs(0,1)] NEW MINER CREATED
[2019-06-08 12:47:47] [INFO] [Miner35-GMiner-algo(Beam)-devs(NVIDIA:2)] Starting miner commandLine='-a 150_5 -s beam.usa.nicehash.com -n 3370 -u 35MQaSQkNoprbFX9yQJsUu3gkk3KAru6qU.worker1 -d 2 -w 0 --api 4003 '
[2019-06-08 12:47:47] [INFO] [Miner35-GMiner-algo(Beam)-devs(NVIDIA:2)] Starting miner environmentVariables='<null>'
[2019-06-08 12:47:47] [INFO] [Miner36-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Starting miner watchdog task
[2019-06-08 12:47:47] [INFO] [MiningManager] Stop Old Mining: (GrinCuckaroo29_BMiner(GPU-c24cc578-6da1-704b-bcb8-c2d5b2cd28f2),GrinCuckaroo29_GMiner(GPU-cc52215e-c6f0-ce01-2b05-087e7ea9d4ec),GrinCuckatoo31_NBMiner(GPU-e7b77b37-a8e6-6205-8742-9c399efe4a6e))
[2019-06-08 12:47:47] [INFO] [MiningManager] Start New Mining : (Beam_GMiner(GPU-cc52215e-c6f0-ce01-2b05-087e7ea9d4ec),GrinCuckatoo31_NBMiner(GPU-c24cc578-6da1-704b-bcb8-c2d5b2cd28f2,GPU-e7b77b37-a8e6-6205-8742-9c399efe4a6e))
[2019-06-08 12:47:47] [INFO] [MiningManager] No change : (EMTPY)
[2019-06-08 12:47:47] [ERROR] [Miner36-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-08 12:47:47] [INFO] [SOCKET] Sending data: {"method":"devices.status","devices":[[0,"Intel(R) Celeron(R) CPU G3930 @ 2.90GHz",2,49,-1,-1],[1,"NVIDIA TITAN Xp",5,99,86,3380],[2,"NVIDIA TITAN Xp",5,98,85,2938],[3,"MSI GeForce GTX 1070 Ti",5,99,64,1035]]}
[2019-06-08 12:47:48] [ERROR] [Miner35-GMiner-algo(Beam)-devs(NVIDIA:2)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-08 12:47:50] [INFO] [Miner36-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Starting miner commandLine='-a cuckatoo -o stratum+tcp://grincuckatoo31.usa.nicehash.com:3372 -u 35MQaSQkNoprbFX9yQJsUu3gkk3KAru6qU.worker1 --api 127.0.0.1:4004 -d 0,1 -RUN '
[2019-06-08 12:47:50] [INFO] [Miner36-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Starting miner environmentVariables='<null>'
[2019-06-08 12:47:51] [ERROR] [Miner36-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-08 12:47:52] [ERROR] [Miner35-GMiner-algo(Beam)-devs(NVIDIA:2)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-08 12:47:57] [ERROR] [Miner36-NBMiner-algo(GrinCuckatoo31)-devs(NVIDIA:0,NVIDIA:1)] Error occured while getting API stats: An error occurred while sending the request.
[2019-06-08 12:48:17] [INFO] [SwitchingManager] Normalizing profits
```

**Comment by S74nk0** [comment]
New version released [1.9.1.5](https://github.com/nicehash/NiceHashMinerLegacy/releases/tag/1.9.1.5). If any of these issues are still present open a new issue.
@F1Rumors please don't recycle an issue thread. For different bugs please open different issues.

Closing
